spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/dsl/StoredProcExecutorConfigurer.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.dsl;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.expression.Expression;
+import org.springframework.integration.jdbc.SqlParameterSourceFactory;
+import org.springframework.integration.jdbc.StoredProcExecutor;
+import org.springframework.integration.jdbc.storedproc.ProcedureParameter;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.SqlParameter;
+
+/**
+ * A {@link StoredProcExecutor} configurer.
+ *
+ * @author Jiandong Ma
+ *
+ * @since 7.0
+ */
+public class StoredProcExecutorConfigurer {
+
+	private final StoredProcExecutor storedProcExecutor;
+
+	protected StoredProcExecutorConfigurer(StoredProcExecutor storedProcExecutor) {
+		this.storedProcExecutor = storedProcExecutor;
+	}
+
+	/**
+	 * @param ignoreColumnMetaData the ignoreColumnMetaData
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setIgnoreColumnMetaData(boolean)
+	 */
+	public StoredProcExecutorConfigurer ignoreColumnMetaData(boolean ignoreColumnMetaData) {
+		this.storedProcExecutor.setIgnoreColumnMetaData(ignoreColumnMetaData);
+		return this;
+	}
+
+	/**
+	 * @param procedureParameters the procedureParameters
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setProcedureParameters(List)
+	 */
+	public StoredProcExecutorConfigurer procedureParameters(List<ProcedureParameter> procedureParameters) {
```

Comment:
Can we have this as vararg to not make end-user to use that `List.of()`?
Or... We can have just `procedureParameter(ProcedureParameter procedureParameter)` API and explain in its JavaDocs that it is safe to call it several times for more params.